src/main/scala/wolfendale/nunjucks/TemplateNode.scala

```diff
@@ -94,6 +96,40 @@ object TemplateNode {
     final case class ConditionalContent(condition: expression.syntax.AST, content: Partial)
   }
 
+
+  final case class Switch(exprToMatch:expression.syntax.AST.Expr, contents: Seq[Switch.ConditionalContent], default: Option[Partial]) extends Tag {
+
+    override def eval: State[Context, String] = {
+
+      val all: Seq[Switch.ConditionalContent] = default
+        .map(contents :+ Switch.ConditionalContent(Seq(exprToMatch), _))
+        .getOrElse(contents)
+
+      val collapsedConditionals = all.foldRight(Seq.empty[Switch.ConditionalContent]){
+        (m, n) => if(m.content.isEmpty){
+          val last = n.head
+          Switch.ConditionalContent(last.condition ++ m.condition, last.content) +: n.tail
+        }else {
+          m +: n
+        }
+      }
+
+      collapsedConditionals.foldLeft(State.pure[Context, Option[String]](None)) { (m, n) =>
+        m.flatMap {
+          case None =>
+            Monad[State[Context, *]].ifM(State.inspect(context => n.condition.exists(_.eval(context).equals(exprToMatch.eval(context)))))(
```

Comment:
Is this the right kind of equality?

We have 4 different kinds:
- `==` / `equals` (standard javascript equality)
- `eq` (referential equality)
- ``==`` (coercive equality for `Value`s) *note: this could probably do with a better name
- ``===`` (strict equality for `Value`s)

Using the following template it suggests that strict equality is used in nunjucks:
```
{% set foo = 0 %}
{% switch foo %}
{% case false %}coercive
{% default %}strict
{% endswitch %}
``` 

outputs `strict`

Reply:
test for this demonstrates strict equality being used